File: jj/faultdiag-source-20260906/annotation-app/frontend/src/api/health.ts
import http from './http'

export interface HealthResponse {
  status: string
  version?: string
  database?: {
    connected: boolean
    error?: string | null
  }
  [key: string]: unknown
}

export async function getHealth() {
  const { data } = await http.get<HealthResponse>('/health')
  return data
}

File: jj/code/annotation-app/backend/app/api/v1/health.py
from fastapi import APIRouter

from app.core.config import get_settings
from app.db.health import check_database
from app.db.session import engine

router = APIRouter(tags=["health"])


@router.get("/health")
def health() -> dict[str, object]:
    settings = get_settings()
    db_ok, db_error = check_database(engine)
    return {
        "status": "ok" if db_ok else "degraded",
        "version": settings.app_version,
        "database": {
            "connected": db_ok,
            "error": db_error,
        },
    }
